Mathlib/Order/Preorder/Chain.lean

```diff
@@ -109,6 +116,66 @@ theorem IsChain.lt_of_le [PartialOrder α] {s : Set α} (h : IsChain (· ≤ ·)
     IsChain (· < ·) s := fun _a ha _b hb hne ↦
   (h ha hb hne).imp hne.lt_of_le hne.lt_of_le'
 
+section Rel
+
+variable {r : α → α → Prop} {r' : β → β → Prop} {s : Set α}
+
+theorem IsChain.image_relEmbedding (hs : IsChain r s) (φ : r ↪r r') : IsChain r' (φ '' s) := by
```

Comment:
The proofs of these first two theorems are slightly adjusted to account for differences betwen IsChain/IsAntichain, all of the following lemmas are exact copy-and-replace.